icc-x-api/icc-crypto-x-api.ts

```diff
@@ -1342,9 +1349,16 @@ export class IccCryptoXApi {
     return this.hcpartyBaseApi
       .getHealthcareParty(hcpId)
       .then(hcp => {
-        const crt = this.getKeychainInBrowserLocalStorageAsBase64(hcp.id!!)
         const opts = hcp.options || {}
+
+        const crt = this.getKeychainInBrowserLocalStorageAsBase64(hcp.id!!)
         _.set(opts, this.hcpPreferenceKeyEhealthCert, crt)
+
+        const crtValidityDate = this.getKeychainValidityDateInBrowserLocalStorage(hcp.id!!)
+        if (opts[this.hcpPreferenceKeyEhealthCertDate] !== crtValidityDate) {
+          _.set(opts, this.hcpPreferenceKeyEhealthCertDate, crtValidityDate)
+        }
```

Comment:
Sorry I missed this in the first round, but if the value does not exist in local storage the outcome will be `null`, which is different from the original `undefined` value of the options. So if the validity is not stored (I'm thinking of other applications that use the ICC API) it will still be subject to change.

Perhaps the getter should replace the `null` produced by the `localStorage` API with an `undefined` to avoid that. This would equalize the behavior of the `localStorage` API and object property access when the property does not exist.

Unless of course you intend to store a `null` value in the `opts` when the validity date is not defined in local storage.